src/sync/impl/sync_metadata.cpp

```diff
@@ -82,18 +82,19 @@ namespace realm {
 
 SyncMetadataManager::SyncMetadataManager(std::string path,
                                          bool should_encrypt,
-                                         util::Optional<std::vector<char>> encryption_key)
+                                         util::Optional<std::vector<char>> encryption_key,
+                                         __unused bool should_recheck_keychain)
```

Comment:
`__unused` is GCC/clang-specific. The pattern core uses for parameters only used on some platforms is a `static_cast<void>(param)` in the `#else` block.

Reply:
Oh, that makes sense. Thanks!